linera-core/src/notifier.rs

```diff
@@ -70,15 +74,37 @@ where
     }
 }
 
-impl Notifier<crate::worker::Notification> {
-    /// Process multiple notifications of type [`crate::worker::Notification`].
-    pub fn handle_notifications(&self, notifications: &[crate::worker::Notification]) {
+impl Notifier<worker::Notification> {
+    /// Process multiple notifications of type [`worker::Notification`].
+    pub fn handle_notifications(&self, notifications: &[worker::Notification]) {
         for notification in notifications {
             self.notify(&notification.chain_id, notification);
         }
     }
 }
 
+pub trait NotificationSink<N: Clone>: Clone + Send + 'static {
+    fn handle_notifications(&self, notifications: &[N]);
```

Comment:
I think it would be better to have this trait extend `Extend` and use the pre-existing `Extend::extend` method. 

(`Extend` can use `Iterator::size_hint` for efficiency, so there shouldn't be a performance benefit to restricting the input type to a concrete slice.)

Reply:
I'm not sure: [`Extend` is specifically meant for collections](https://doc.rust-lang.org/std/iter/trait.Extend.html), and the notifier is not a collection.

Reply:
Makes sense to me :)